### PR TITLE
Use enum for message kind in compiletest harness.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: src/compiletest/errors.rs [hunk 1]
```diff
@@ -84,8 +124,9 @@ fn parse_expected(last_nonfollow_error: Option<usize>,
     let letters = line[kind_start..].chars();
     let kind = letters.skip_while(|c| c.is_whitespace())
                       .take_while(|c| !c.is_whitespace())
```
Comment (nikomatsakis) [inline src/compiletest/errors.rs]: Surely we have some higher-level way to extract the first word from the string by now? Can't we just call `split_whitespace()` or something?

Comment (frewsxcv) [inline src/compiletest/errors.rs]: Addressed in https://github.com/rust-lang/rust/pull/32341/commits/73f432189298dd07195427fcc9216ed2c5ba8d11

File: src/compiletest/errors.rs [hunk 2]
```diff
@@ -121,12 +121,11 @@ fn parse_expected(last_nonfollow_error: Option<usize>,
...
     };
     let kind_start = start + tag.len() + adjusts + (follow as usize);
-    let letters = line[kind_start..].chars();
-    let kind = letters.skip_while(|c| c.is_whitespace())
-                      .take_while(|c| !c.is_whitespace())
-                      .collect::<String>()
-                      .parse::<ErrorKind>()
-                      .ok();
+    let kind = line[kind_start..].split_whitespace()
+                                 .next()
```
Comment (nikomatsakis) [inline src/compiletest/errors.rs]: much better :)

**Pull request merged**